text/0000-eye-of-sauron.md

```diff
@@ -0,0 +1,256 @@
+- Feature Name: eye_of_sauron
+- Start Date: 2017-09-11
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow for method-dispatch-style trait-driven autoderef and autoref in operators.
+
+# Motivation
+[motivation]: #motivation
+
+In today's Rust, working with operators is annoying, because they are supposed to be a "lightweight" syntax like methods and field accesses, but unlike them, they do not have coercions on their LHS and therefore require explicit autoderefs and autorefs.
+
+One common example where this is a nuisance is when using iterator adapters, working which can easily create multiple-reference types such as `&&u32`. For example:
+    
+```Rust
+    let v : Vec<u32> = vec![0, 1, 2, 3];
+    // Why can't I just write `x > 1`? Why do I have to ** for the compiler???
+    v.iter().filter(|x| **x > 1).count();
+```
+
+There are several cases where these operators are used. The most popular case is indexing associative maps:
+
+```Rust
+    use std::collections::HashMap;
+    let x: HashMap<_, _> =
+        vec![(format!("hello"), format!("world"))].into_iter().collect();
+    let s = format!("hello");
+    
+    // I would like to write...
+    println!("{}", x[s]);
+    // But instead, I have to write...
+    println!("{}", x[&s]);
+```
+
+In this case, these are merely annoying papercuts. In some other cases, they can be a much worse problem.
+
+One of thim is Isis Lovecruft's "Eye of Sauron" case, which is a problem when using non-`Copy` bignums:
```

Comment:
`s/thim/them/`?

BTW I think linking to https://twitter.com/isislovecruft/status/839333846415998976 may provide some context what is meant by *Isis Lovecruft's "Eye of Sauron" case*.

Reply:
Just to nit pick, there are no bignums in curve25519, only 256bit numbers, which do not count as bignums because they do not use `Box<>`.  Also, those numbers could easily be `Copy` so if they are not it's presumably Isis and Henry found that 32 bytes is already big enough for `Copy` to cause a performance penalty